lib/Backends/OpenCL/OpenCL.cpp

```diff
@@ -607,13 +555,25 @@ static void topK(Tensor &outW, Tensor &indW, Tensor &inW, size_t k) {
 void OpenCLFunction::execute(ExecutionContext *context) {
   (void)context;
 
-  {
-    auto ev = context->scopedEvent("alloc");
-    // TODO: deviceBuffer_ should go into DeviceBindings in context.
-    deviceBuffer_ = allocDeviceBuffer(runtimeBundle_.getConstantWeightSize() +
-                                      runtimeBundle_.getMutableWeightSize() +
-                                      runtimeBundle_.getActivationsSize());
-  }
+  auto clBindings = static_cast<runtime::OpenCLDeviceBindings *>(
+      context->getDeviceBindings());
+
+  deviceBuffer_ = clBindings->deviceBuffer;
```

Comment:
Do you assign to the fields of `OpenCLFunction` here? This is not thread-safe, or? We may run the same `OpenCLFunction` on different inputs concurrently. 

Shouldn't these fields be removed from `OpenCLFunction` now that they are in the `OpenCLDeviceBindings`?

Reply:
That is the next step. This PR introduces the bindings and moves allocation and setup out of the function. The next PR would be a refactor passing the bindings around to the functions that currently rely on the member variables.